### ISSUE TITLE
Platform Support

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/feedback#174]
Support for platforms outside of macOS is planned, but it is not being prioritized in the short term (nor is it currently being tracked in the [Top-Ranking Issues](https://github.com/zed-industries/feedback/issues/52)).  Feel free to add 👍s and discuss anything you might want in the following locations:

- https://github.com/zed-industries/feedback/issues/171
- https://github.com/zed-industries/feedback/issues/172
- https://github.com/zed-industries/feedback/issues/173

**Comment by aminya** [comment]
I tried to run it inside a Catalina VM because I don't use a Mac. That's unfortunate, as I really wanted to test Zed. 

First, I could not run it from the image file itself. After realizing I needed to move it to the Application folder, I reran it multiple times, but it failed to run.

Then, by running it from the terminal I could get an error message:

```
Error gpui::platform::mac::window] Unable to access a compatible graphics device
```
![image](https://user-images.githubusercontent.com/16418197/185080400-2ea8367a-f01f-4b5e-9619-15d9c5feecef.png)

I installed the VM using Sosumi snap on Kubuntu 22.04.

This seems similar to 
https://github.com/zed-industries/feedback/issues/272

**Comment by iamnbutler** [comment]
> I tried to run Zed inside a Catalina VM because I don't use a Mac, however it failed. That's unfortunate, as I really wanted to test Zed.
> 
> First, I could not run it from the image file itself. After realizing that I needed to move it to the Application folder, I reran it multiple times, but it failed to run without showing any message.
> 
> Then, by running it from the terminal I could get an error message:
> 
> ```
> Error gpui::platform::mac::window] Unable to access a compatible graphics device
> ```
> 
> ![image](https://user-images.githubusercontent.com/16418197/185080400-2ea8367a-f01f-4b5e-9619-15d9c5feecef.png)
> 
> I installed the VM using Sosumi snap on Kubuntu 22.04.
> 
> This seems similar to #272

Yeah unfortunately you won't be able to run Zed in a VM. We added that message as previously zed would try to open and immediately panic. I don't remember where, but someone tried this before and had an issue for it. 

One of the other team members will need to speak to the technical side, but I think it has something to do with Metal.

**Comment by maxbrunsfeld** [comment]
Yeah, when running macOS on a VM, we don't have the GPU access that we currently depend on for rendering.

**Comment by iamnbutler** [comment]
_Reposting my discord response about platform support here:_

Hey y'all – We are excited that you all are excited to try Zed. Just want to make sure expectations are in check – We plan to release on multiple platforms for 1.0, but work on these has not even begun yet. 

We're focusing on getting the editor into a good place on one platform first, then will direct attention to making it more widely available. We do have some team members hacking on experimental implementations on other platforms, but they are not official currently and have no expected date they will be testable.

We want to make sure that expectations are clear – We know y'all are dying to try Zed out on the platforms you love the most. I try to pop in when I can to make sure expectations are in check, but I've been a bit heads down recently.

From our faq (https://zed.dev/faq):

> We're focusing on a single platform for now because we're a small team. At our current scale, maintaining additional platforms would represent a big cost without much additional learning. We plan to support Linux and Windows before 1.0, and to support a web version some time after that.

There is a tracking issue for platform support that you can follow along with here: https://github.com/zed-industries/feedback/issues/174 (this issue)

**Comment by heedfull** [comment]
FYI The windows issue is now locked, so I can't thumbs up to vote for it

**Comment by patmaddox** [comment]
Please consider targeting FreeBSD as well. I haven't developed X11 apps on it myself, but [there is a robust collection of X11 apps in the ports tree](https://cgit.freebsd.org/ports/tree/x11).

**Comment by patmaddox** [comment]
> One would hope that a new project like this pushing the cutting edge of GUI tech would skip it and go straight to targeting Wayland.

I haven’t used it, but [FreeBSD supports Wayland](https://docs.freebsd.org/en/books/handbook/wayland/). There’s a lot of [user discussion about Wayland](https://forums.freebsd.org/search/311677/?q=wayland&o=date), so I guess support is pretty good and there will be continued investment in it.

It looks like the ports are keeping up with Wayland releases:

- https://www.freshports.org/graphics/wayland/
- https://www.freshports.org/graphics/wayland-protocols

**Comment by blurayne** [comment]
Same time of support for Linux like announced by Warp? Then you better shouldn't even put Linux support on your list. 

Zed just died for me.

**Comment by w1hm1** [comment]
So, Zed is for posers. Thank you – now I know what it is all about pretty much.

**Comment by mlevkov** [comment]
Is there a reason why Zed is only supporting 10.15+ and not versions prior? I'm on 10.14 and cannot update to 10.15. However, I would be interested in trying Zed.

**Comment by JosephTLyons** [comment]
> Is there a reason why Zed is only supporting 10.15+ and not versions prior? I'm on 10.14 and cannot update to 10.15. However, I would be interested in trying Zed.

[A feature we added required a library that made us bump our OS minimum version requirement](https://docs.zed.dev/general/system-requirements)

**Comment by mlevkov** [comment]
Making it an optional plug-in would allow users who do not need such a feature to consider using your IDE. I, for example, care less about sharing code with others in either pair programming sessions or otherwise. However, I'd be delighted to use the editor for my needs, but I will not change the os and the setup to use the editor. Thus, a delayed consideration may occur here once the OS change occurs.

**Comment by ZZzzaaKK** [comment]
> Currently, Zed only supports macOS. Support for other platforms is planned, but it is not being prioritized in the short term.
> 
> Feel free to add +1s and discuss anything you might want in the following issues:
> 
> * https://github.com/zed-industries/feedback/issues/172
> * https://github.com/zed-industries/feedback/issues/171
> * https://github.com/zed-industries/feedback/issues/173
> 
> Updates to development in cross-compatibility will be posted to these individual issues, whenever work in that area begins.
> 
> Note: We are currently not tracking platform support issues in the [Top-Ranking Issues](https://github.com/zed-industries/feedback/issues/52), in order to keep the list geared more towards short-term items.

The links to the platform-specific issues appear to be broken. Could you replace ``feedback`` with ``community`` in each link to make them valid again like so:
https://github.com/zed-industries/community/issues/172
https://github.com/zed-industries/community/issues/171
https://github.com/zed-industries/community/issues/173

**Comment by hovsater** [comment]
@ZZzzaaKK thanks for the heads up. I've updated the links. 🙂
